src/freenet/client/filter/CSSTokenizerFilter.java

```diff
@@ -1773,26 +1773,26 @@ else if(HTMLelement.indexOf('#')!=-1)
 			}
 
 		}
-		if(isIDSelector && "".equals(id)) return null; // No ID
+		if(isIDSelector && id.isEmpty()) return null; // No ID
```

Comment:
are we sure here that id cannot be null? the check before was null-safe.

Reply:
Yes, this is either an empty string (assigned at declaration), or it’s the output of a `trim()` method call so this cannot be `null`.